### PR TITLE
feat: call onNewNftEvent when a new NFT tx is received

### PR DESCRIPTION
**Pull request by andreabadesso**
### Motivation

This fixes https://github.com/HathorNetwork/hathor-wallet-service/issues/153

After the wallet-service refactor, we're no longer calling the old `onNewTxRequest` lambda (as what it used to do is now done by the daemon), so the logic that deals with NFTs should be called from the daemon.

This PR is not attempting to refactor the wallet-service and move all shared code to the `common` package, but to move only the minimum necessary code so that the nft util works both on the wallet-service and the daemon package.

### Acceptance Criteria

- We should move the nft utils from the wallet-service repository to the `common` package, that has methods common to both the lambdas and the daemon
- We should move all transaction types to the `common` package
- We should call use the nft util methods from the `common` package to call the `onNewNftEvent` lambda in the `newVertexAccepted` service of the daemon

### Checklist
- [X] If you are requesting a merge into `master`, confirm this code is production-ready and can be included in future releases as soon as it gets merged
- [X] Make sure either the unit tests and/or the QA tests are capable of testing the new features
- [X] Make sure you do not include new dependencies in the project unless strictly necessary and do not include dev-dependencies as production ones. More dependencies increase the possibility of one of them being hijacked and affecting us.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -3,15 +3,16 @@
...
+    "packages/common"
   ],
   "engines": {
     "node": ">=18"
   },
   "nohoist": [
     "**"
   ],
-  "repository": "git@github.com:HathorNetwork/hathor-wallet-service-sync_daemon.git",
+  "repository": "git@github.com:HathorNetwork/hathor-wallet-service.git",
```
Comment (andreabadesso) [inline package.json]: I renamed the repository

File: packages/common/package.json [hunk 2]
```diff
@@ -0,0 +1,12 @@
+{
+  "name": "@wallet-service/common",
+  "packageManager": "yarn@4.1.0",
+  "peerDependencies": {
+    "@aws-sdk/client-lambda": "3.540.0",
+    "@hathor/wallet-lib": "0.39.0",
+    "winston": "^3.13.0"
```
Comment (andreabadesso) [inline packages/common/package.json]: These are defined as `peerDependencies` so they use the same version from the root `package.json`

File: packages/daemon/src/types/event.ts [hunk 3]
```diff
@@ -75,6 +75,7 @@ export type FullNodeEvent = {
       tokens: string[];
       token_name: null | string;
       token_symbol: null | string;
+      signal_bits: number;
```
Comment (andreabadesso) [inline packages/daemon/src/types/event.ts:78]: We should open a PR to start storing this in the transactions table, but this is not yet coming from the event stream.

File: packages/daemon/src/services/index.ts [hunk 4]
```diff
@@ -343,9 +347,23 @@ export const handleVertexAccepted = async (context: Context, _event: Event) => {
...
+      // Validating for NFTs only after the tx is successfully added
+      if (NftUtils.shouldInvokeNftHandlerForTx(tx, network, logger)) {
+        // This process is not critical, so we run it in a fire-and-forget manner, not waiting for the promise.
+        // In case of errors, just log the asynchronous exception and take no action on it.
+        NftUtils.invokeNftHandlerLambda(tx.tx_id, STAGE)
+          .catch((err) => logger.error('[ALERT] Errored on nftHandlerLambda invocation', err));
+      }
     }
 
-    // TODO: Send message on SQS  for real-time update
```
Comment (andreabadesso) [inline packages/daemon/src/services/index.ts:348]: We already did this in the past, it's on line 327

File: packages/common/__tests__/utils/nft.utils.test.ts [hunk 5]
```diff
@@ -128,7 +152,8 @@ describe('isTransactionNFTCreation', () => {
 
     // Validation
     const tx = getTransaction();
-    const result = NftUtils.isTransactionNFTCreation(tx);
+    // @ts-ignore
```
Comment (tuliomir) [inline packages/common/__tests__/utils/nft.utils.test.ts]: question: Is this flag still needed?

File: packages/common/package.json [hunk 6]
```diff
@@ -1,12 +1,20 @@
 {
   "name": "@wallet-service/common",
   "packageManager": "yarn@4.1.0",
+  "scripts": {
+    "jest": "jest --runInBand --collectCoverage --detectOpenHandles --forceExit",
+    "test": "jest"
```
Comment (tuliomir) [inline packages/common/package.json]: question(non-blocking): Do we have any benefit of having this `jest` script instead of just a `test` one?

File: packages/common/src/types.ts
Comment (tuliomir) [inline packages/common/src/types.ts:1]: We should have the Hathor header on this file.

Also, this is a types file that is half filled with _utils_ functions. Maybe this is not the time to extract all those functions into separate dedicated files, but I believe at least an issue would be important to do so. In the future we may have circular dependencies because of these mixed responsibilities.

File: packages/common/__tests__/utils/nft.utils.test.ts (hunk 5 above)
Comment (andreabadesso) [reply packages/common/__tests__/utils/nft.utils.test.ts]: It's not, removed it in c92fafac7661553185d9992a25f409169d59b4e2

File: packages/common/package.json (hunk 6 above)
Comment (andreabadesso) [reply packages/common/package.json]: We don't... Removed the jest script in 13f98b61cca8c4b3f6f5af88dacc189fe17832c9

File: packages/common/src/types.ts
Comment (andreabadesso) [reply packages/common/src/types.ts:1]: Adde the header in 61fe30e710758bce5e9ca74fc6b41a54990022c6


I'm writing a project listing all refactors I think we need to do to share code between projects, I'll add this there

**Review by tuliomir — APPROVED** [review]

File: packages/common/__tests__/events/nftCreationTx.ts [hunk 7]
```diff
@@ -0,0 +1,181 @@
+/**
```
Comment (pedroferreira1) [inline packages/common/__tests__/events/nftCreationTx.ts]: Should we delete the file in the wallet service package that is exactly this one?

https://github.com/HathorNetwork/hathor-wallet-service/blob/master/packages/wallet-service/events/nftCreationTx.ts

File: packages/wallet-service/src/txProcessor.ts [hunk 8]
```diff
@@ -81,6 +81,8 @@ export const IGNORE_TXS = {
   ],
 };
 
+export const CREATE_NFT_MAX_RETRIES = 5;
```
Comment (pedroferreira1) [inline packages/wallet-service/src/txProcessor.ts]: We used to have an env var for this, why did you change it?

```
export const MAX_METADATA_UPDATE_RETRIES: number = parseInt(process.env.MAX_METADATA_UPDATE_RETRIES || '3', 10);
```

File: packages/common/src/utils/nft.utils.ts [hunk 9]
```diff
@@ -26,24 +23,30 @@ export class NftUtils {
   /**
    * Returns whether we should invoke our NFT handler for this tx
    * @param {Transaction} tx
+   * @param {string} network
```
Comment (pedroferreira1) [inline packages/common/src/utils/nft.utils.ts]: Missing `logger` parameter and wrong `network` type. Since it's a typescript file I think we could omit the types in the docstring.

File: packages/common/src/utils/nft.utils.ts [hunk 10]
```diff
@@ -124,29 +127,31 @@ export class NftUtils {
   /**
    * Identifies if the metadata for a NFT needs updating and, if it does, update it.
    * @param {string} nftUid
+   * @param {number} maxRetries
```
Comment (pedroferreira1) [inline packages/common/src/utils/nft.utils.ts]: Missing `logger` parameter

File: packages/wallet-service/src/txProcessor.ts (hunk 8 above)
Comment (andreabadesso) [reply packages/wallet-service/src/txProcessor.ts]: Refactored it in 4448e29f567e858ebbc5950d461e5ed366ce0289

We were not using this env, it was always defaulting to 3, added it to codebuild as well

File: packages/common/src/utils/nft.utils.ts (hunk 9 above)
Comment (andreabadesso) [reply packages/common/src/utils/nft.utils.ts]: Fixed in 16efa8fdcf591e6e3c8b74a7e8e9e7bcb2f2d9e2

File: packages/common/src/utils/nft.utils.ts (hunk 10 above)
Comment (andreabadesso) [reply packages/common/src/utils/nft.utils.ts]: Fixed in 16efa8fdcf591e6e3c8b74a7e8e9e7bcb2f2d9e2

File: packages/common/__tests__/events/nftCreationTx.ts (hunk 7 above)
Comment (andreabadesso) [reply packages/common/__tests__/events/nftCreationTx.ts]: Yes, removed in 25bb3a319d0606b08076e033720f6b493e2d56b0

**Review by pedroferreira1 — APPROVED** [review]
